lex.go

```diff
@@ -226,3 +318,146 @@ func tokenize(reader io.Reader, tokenCh chan NgxToken) {
 
 	close(tokenCh)
 }
+
+type LuaLexer struct {
+	s LexScanner
+}
+
+func (ll *LuaLexer) Register(s LexScanner) []string {
+	ll.s = s
+	return []string{
+		"init_by_lua_block",
+		"init_worker_by_lua_block",
+		"exit_worker_by_lua_block",
+		"set_by_lua_block",
+		"content_by_lua_block",
+		"server_rewrite_by_lua_block",
+		"rewrite_by_lua_block",
+		"access_by_lua_block",
+		"header_filter_by_lua_block",
+		"body_filter_by_lua_block",
+		"log_by_lua_block",
+		"balancer_by_lua_block",
+		"ssl_client_hello_by_lua_block",
+		"ssl_certificate_by_lua_block",
+		"ssl_session_fetch_by_lua_block",
+		"ssl_session_store_by_lua_block",
+	}
+}
+
+//nolint:funlen,gocognit,gocyclo,nosec
+func (ll *LuaLexer) Lex(matchedToken string) <-chan NgxToken {
+	tokenCh := make(chan NgxToken)
+
+	tokenDepth := 0
+
+	go func() {
+		defer close(tokenCh)
+		var tok strings.Builder
+		var inQuotes bool
+
+		// special handling for'set_by_lua_block' directive
+		// #nosec G101
+		if matchedToken == "set_by_lua_block" { //nolint
+			arg := ""
+			for {
+				if !ll.s.Scan() {
+					return
+				}
+				next := ll.s.Text()
+				if isSpace(next) {
+					if arg != "" {
+						tokenCh <- NgxToken{Value: arg, Line: ll.s.Line(), IsQuoted: false}
+						break
+					}
+
+					for isSpace(next) {
+						if !ll.s.Scan() {
+							return
+						}
+						next = ll.s.Text()
+					}
+				}
+				arg += next
+			}
+			// skip leading whitespace after the return value
```

Comment:
Why is this nested in the block to handle the `set_by_lua_block` token? In the Python implementation whitespace is consumed looking for the Lua block to start with `{` regardless of the matched token.

https://github.com/nginxinc/crossplane/blob/master/crossplane/ext/lua.py#L77

Reply:
Done